File: POSmaisgestao/app/src/main/java/com/pos_mais_gestao/domain/balanca/BalancaLeitura.java
package com.pos_mais_gestao.domain.balanca;

import java.math.BigDecimal;

public final class BalancaLeitura {
    private final BalancaEstado estado;
    private final BigDecimal pesoKg;
    private final String mensagem;

    public BalancaLeitura(BalancaEstado estado, BigDecimal pesoKg, String mensagem) {
        this.estado = estado;
        this.pesoKg = pesoKg;
        this.mensagem = mensagem;
    }

    public static BalancaLeitura estado(BalancaEstado estado, String mensagem) {
        return new BalancaLeitura(estado, null, mensagem);
    }

    public static BalancaLeitura peso(BigDecimal pesoKg) {
        return new BalancaLeitura(BalancaEstado.ESTAVEL, pesoKg, "Peso estável");
    }

    public BalancaEstado getEstado() {
        return estado;
    }

    public BigDecimal getPesoKg() {
        return pesoKg;
    }

    public String getMensagem() {
        return mensagem;
    }

    public boolean isPesoValido() {
        return estado == BalancaEstado.ESTAVEL
                && pesoKg != null
                && pesoKg.compareTo(BigDecimal.ZERO) > 0;
    }
}

File: estacao-balanca/lib/features/balanca/balanca_leitura.dart
enum BalancaEstado {
  desabilitada,
  desconectada,
  conectando,
  conectada,
  lendo,
  estavel,
  instavel,
  pesoNegativo,
  sobrecarga,
  erro,
}

class BalancaLeitura {
  const BalancaLeitura({
    required this.estado,
    this.pesoKg,
    this.mensagem,
  });

  final BalancaEstado estado;
  final double? pesoKg;
  final String? mensagem;

  factory BalancaLeitura.estado(BalancaEstado estado, String mensagem) {
    return BalancaLeitura(estado: estado, mensagem: mensagem);
  }

  factory BalancaLeitura.peso(double pesoKg) {
    return BalancaLeitura(
      estado: BalancaEstado.estavel,
      pesoKg: pesoKg,
      mensagem: 'Peso estável',
    );
  }

  bool get isPesoValido =>
      estado == BalancaEstado.estavel &&
      pesoKg != null &&
      pesoKg! > 0;
}
